### PR TITLE
Enhance Users controller and LDAP logic

### LINKED ISSUE #172: Non-admins should be able to view users
In order to facilitate users being able to administer their own projects, users should be able to lookup other users without having admin privileges.

### PR DESCRIPTION
**Pull request by Enquier**
This is a proposed fix for #172 which allows non-admins to lookup users. It also adds a query parameter which to allow search for a specific username.

### CODE REVIEW

File: ldap/src/main/java/org/openmbee/mms/ldap/LdapSecurityConfig.java [hunk 1]
```diff
@@ -178,4 +189,28 @@ public BaseLdapPathContextSource contextSource() {
...
+        }
+        if (!saveUser.getFirstName().equals(userData.getStringAttribute(userAttributesFirstName))) {
+            saveUser.setFirstName(userData.getStringAttribute(userAttributesFirstName));
+        }
+        if (!saveUser.getLastName().equals(userData.getStringAttribute(userAttributesLastName))) {
+            saveUser.setLastName(userData.getStringAttribute(userAttributesLastName));
+        }
+
+        saveUser.setEnabled(true);
+        saveUser.setAdmin(false);
```
Comment (HuiJun) [inline ldap/src/main/java/org/openmbee/mms/ldap/LdapSecurityConfig.java]: These will need to be considered. We don't want to modify an admin user and have this logic convert them to a regular user and we don't want to enable a disabled user on modification either.

File: build.gradle [hunk 2]
```diff
@@ -157,4 +157,37 @@ subprojects {
             sign publishing.publications.mavenJava
         }
     }
-}
\ No newline at end of file
+
+}
+
```
Comment (HuiJun) [inline build.gradle]: Why is this?

**Review by HuiJun — CHANGES_REQUESTED** [review]

File: ldap/src/main/java/org/openmbee/mms/ldap/LdapSecurityConfig.java (hunk 1 above)
Comment (Enquier) [reply ldap/src/main/java/org/openmbee/mms/ldap/LdapSecurityConfig.java]: Fixed. Moved over to the create side only

File: build.gradle (hunk 2 above)
Comment (Enquier) [reply build.gradle]: Adds ability to run MMS as a server from gradle directly. For rapid development without generating a new docker container for each run

Comment (HuiJun) [reply build.gradle]: MMS already has the ability to run from gradle directly. It's called ./gradlew bootRun.

**Review by HuiJun — CHANGES_REQUESTED** [review]

**Review by HuiJun — APPROVED** [review]

**Pull request merged**